### PR TITLE
[O2B-1151] Auto-fill tags in on-call log template

### PR DESCRIPTION
**Pull request by martinboulais**
#### I have a JIRA ticket
- [X] branch and/or PR name(s) include(s) JIRA ID
- [X] issue has "Fix version" assigned
- [X] issue "Status" is set to "In review"
- [X] PR labels are selected

Notable changes for users:
- Tags are automatically added when filling on-call template

### CODE REVIEW

File: lib/public/components/runTypes/RunTypesSelectionDropdownModel.js [hunk 1]
```diff
@@ -34,10 +34,10 @@ export class RunTypesSelectionDropdownModel extends SelectionDropdownModel {
     _initialize() {
         runTypesProvider.getAll().then(
             (runTypes) => {
-                this.availableOptions = RemoteData.success(runTypes.map(runTypeToOption));
+                this.setAvailableOptions(RemoteData.success(runTypes.map(runTypeToOption)));
             },
             (errors) => {
-                this.availableOptions = RemoteData.failure(errors);
+                this.setAvailableOptions(RemoteData.failure(errors));
             },
```
Comment (xsalonx) [inline lib/public/components/runTypes/RunTypesSelectionDropdownModel.js:41]: not just
```
            (runTypes) => this.setAvailableOptions(RemoteData.success(runTypes.map(runTypeToOption))),
            (errors) => this.setAvailableOptions(RemoteData.failure(errors)),
```

File: lib/public/components/tag/TagPickerModel.js [hunk 2]
```diff
@@ -34,10 +34,10 @@ export class TagPickerModel extends PickerModel {
         const { includeArchived = false } = configuration || {};
         (includeArchived ? tagsProvider.getAll() : tagsProvider.getNotArchived()).then(
             (tags) => {
-                this.availableOptions = RemoteData.success(tags.map(tagToOption));
+                this.setAvailableOptions(RemoteData.success(tags.map(tagToOption)));
             },
             (errors) => {
-                this.availableOptions = RemoteData.failure(errors);
+                this.setAvailableOptions(RemoteData.failure(errors));
             },
```
Comment (xsalonx) [inline lib/public/components/tag/TagPickerModel.js:41]: same as https://github.com/AliceO2Group/Bookkeeping/pull/1409/files#r1506154236

File: lib/public/components/tag/TagSelectionDropdownModel.js [hunk 3]
```diff
@@ -38,10 +38,10 @@ export class TagSelectionDropdownModel extends SelectionDropdownModel {
     _initialize() {
         (this._includeArchived ? tagsProvider.getAll() : tagsProvider.getNotArchived()).then(
             (tags) => {
-                this.availableOptions = RemoteData.success(tags.map(tagToOption));
+                this.setAvailableOptions(RemoteData.success(tags.map(tagToOption)));
             },
             (errors) => {
-                this.availableOptions = RemoteData.failure(errors);
+                this.setAvailableOptions(RemoteData.failure(errors));
             },
```
Comment (xsalonx) [inline lib/public/components/tag/TagSelectionDropdownModel.js:45]: https://github.com/AliceO2Group/Bookkeeping/pull/1409/files#r1506154236

File: lib/public/components/common/selection/SelectionModel.js [hunk 4]
```diff
@@ -154,27 +169,32 @@ export class SelectionModel extends Observable {
...
+            ? this._availableOptions.apply({ Success: addDefaultToOptions })
+            : addDefaultToOptions(this._availableOptions);
     }
 
     /**
      * Defines the list of available options
      *
-     * @param {RemoteData|SelectionOption[]} availableOptions the new available options
+     * @param {RemoteData<SelectionOption[], *>|SelectionOption[]} availableOptions the new available options
+     * @return {void}
```
Comment (xsalonx) [inline lib/public/components/common/selection/SelectionModel.js:195]: It's not quite true, as actual `availableOptions` are those plus defaults

File: lib/public/components/runTypes/RunTypesSelectionDropdownModel.js (hunk 1 above)
Comment (martinboulais) [reply lib/public/components/runTypes/RunTypesSelectionDropdownModel.js:41]: Indeed, right

File: lib/public/components/tag/TagPickerModel.js (hunk 2 above)
Comment (martinboulais) [reply lib/public/components/tag/TagPickerModel.js:41]: Done

File: lib/public/components/tag/TagSelectionDropdownModel.js (hunk 3 above)
Comment (martinboulais) [reply lib/public/components/tag/TagSelectionDropdownModel.js:45]: Done

File: lib/public/components/common/selection/SelectionModel.js (hunk 4 above)
Comment (martinboulais) [reply lib/public/components/common/selection/SelectionModel.js:195]: Available options are the options that are actually available, not the options that are displayed to the user. Again, my english is probably lacking here

**Review by xsalonx — DISMISSED** [review]

**Review by graduta — CHANGES_REQUESTED** [review]
`SL` which stands for `Shift Leader` is not correctly filling in the tag.
Perhaps we should use in the dropdown `Shift Leader` so that the tag is matched

**Review by graduta — APPROVED** [review]

**Pull request merged**